### PR TITLE
Add types to sites

### PR DESCRIPTION
**Pull request by sndrs**
## What does this change?

add types to `sites`

## Why?

types are tops

### CODE REVIEW

File: sites/frontend/components/Header/Nav/SubNav/SubNavList/SubNavListItem/SubNavButton.js [hunk 1]
```diff
@@ -59,7 +61,13 @@ const SubNavButton = styled('button')(
...
 SubNavButton.displayName = 'SubNavButton';
 
-export default ({ pillar, showSecondaryNav, toggleSecondaryNav }) => {
+type Props = {
+    pillar: PillarType,
+    showSecondaryNav: boolean,
+    toggleSecondaryNav: Function,
+};
+
+export default ({ pillar, showSecondaryNav, toggleSecondaryNav }: Props) => {
```
Comment (SiAdcock) [inline sites/frontend/components/Header/Nav/SubNav/SubNavList/SubNavListItem/SubNavButton.js]: Do we want to add return types (`React.Node`) to stateless components? I guess it makes sense

File: sites/frontend/components/Header/Nav/SubNav/SubNavList/SubNavListItem/SubNavButton.js [hunk 2]
```diff
@@ -59,7 +61,13 @@ const SubNavButton = styled('button')(
 );
 SubNavButton.displayName = 'SubNavButton';
 
-export default ({ pillar, showSecondaryNav, toggleSecondaryNav }) => {
+type Props = {
+    pillar: PillarType,
+    showSecondaryNav: boolean,
+    toggleSecondaryNav: Function,
```
Comment (SiAdcock) [inline sites/frontend/components/Header/Nav/SubNav/SubNavList/SubNavListItem/SubNavButton.js]: Maybe make this more explicit

```
toggleSecondaryNav: () => void
```

Flow docs [warn against](https://flow.org/en/docs/types/functions/#toc-function-type) using the `Function` type as it almost completely opts out of the type checker

Comment (sndrs) [reply sites/frontend/components/Header/Nav/SubNav/SubNavList/SubNavListItem/SubNavButton.js]: i just read that actually - nice one

File: sites/frontend/components/Header/Nav/SubNav/SubNavList/SubNavListItem/SubNavButton.js (hunk 1 above)
Comment (sndrs) [reply sites/frontend/components/Header/Nav/SubNav/SubNavList/SubNavListItem/SubNavButton.js]: it ought to be able to infer that shoudn' tit?

Comment (SiAdcock) [reply sites/frontend/components/Header/Nav/SubNav/SubNavList/SubNavListItem/SubNavButton.js]: Yeah I guess. So do we have rules around what types to make explicit and what's safe to assume the compiler can infer?

Comment (SiAdcock) [reply sites/frontend/components/Header/Nav/SubNav/SubNavList/SubNavListItem/SubNavButton.js]: If a function doesn't have any branching, it doesn't need to specify a return type?

**Pull request merged**